app/controllers/auth/MergePageAuthAction.scala

```diff
@@ -55,37 +59,23 @@ class MergePageAuthActionImpl @Inject()(
               .map(key => Uar(key.value))
           }
 
-          val isAgentActive: Boolean = enrolments.find(_.key == "IR-SA-AGENT").map(_.isActivated).getOrElse(false)
-
-          if (saUtr.isEmpty && nino.isEmpty && agentRef.isEmpty) {
-            Future.successful(Redirect(controllers.routes.ErrorController.notAuthorised))
-          } else {
-
-            val authenticatedRequest = AuthenticatedRequest(
-              externalId,
-              agentRef,
-              saUtr.map(SaUtr(_)),
-              nino.map(Nino(_)),
-              saUtr.nonEmpty,
-              isAgentActive,
-              confidenceLevel,
-              credentials,
-              request
-            )
-
-            if (agentRef.isDefined && !isAgentActive) {
-              Future(Redirect(controllers.routes.ErrorController.notAuthorised))
-            } else if (saUtr.isEmpty && agentRef.isEmpty) {
-              nino
-                .map { n =>
-                  handleResponse(authenticatedRequest, n).flatMap(
-                    response => block(response)
-                  )
-                }
-                .getOrElse(block(authenticatedRequest))
-            } else {
-              block(authenticatedRequest)
-            }
+          val isAgentActive: Boolean = enrolments.find(_.key == "IR-SA-AGENT").exists(_.isActivated)
```

Comment:
Returns true if this option is nonempty and the predicate p returns true when applied to this Option's value. Otherwise, returns false.
This is equivalent to:
 option match {
   case Some(x) => p(x)
   case None    => false
 }
 
Params:
p – the predicate to test